addon/-private/system/relationships/belongs-to.js

```diff
@@ -68,6 +68,31 @@ import { DEBUG } from '@glimmer/env';
 
   will lookup for a Post type.
 
+  #### Sync relationships
+
+  Ember Data resolves sync relationships with the related resources
+  available in its local store, hence it is expected these resources
+  to be loaded before or along-side the primary resource.
+
+  ```app/models/comment.js
+  import DS from 'ember-data';
+
+  export default DS.Model.extend({
+    post: DS.belongsTo('post', {
+      async: false
+    })
+  });
+  ```
+
+  In contrast to async relationship, accessing a sync relationship
+  will never trigger a request to fetch the resource,
```

Comment:
This isn't strictly 100% true, but it is "true enough". There are situations involving `reload` in which we won't error and will fire a request. 

Perhaps a better wording would be `will error on access when a related resource is known to exist but has not been loaded`.